plugins/woocommerce/src/Admin/Features/ProductBlockEditor/RedirectionController.php

```diff
@@ -62,8 +69,17 @@ protected function is_legacy_edit_screen(): bool {
 	 * @param integer $product_id Product ID.
 	 */
 	protected function is_product_supported( $product_id ): bool {
-		$product = $product_id ? wc_get_product( $product_id ) : null;
-		$digital_product = $product->is_downloadable() || $product->is_virtual();
+		$product             = $product_id ? wc_get_product( $product_id ) : null;
+		$digital_product     = $product->is_downloadable() || $product->is_virtual();
+		$product_template_id = $product->get_meta( '_product_template_id' );
+
+		if ( isset( $product_template_id ) ) {
+			foreach ( $this->product_templates as $product_template ) {
+				if ( $product_template_id === $product_template->get_id() && ! is_null( $product_template->get_layout_template_id() ) ) {
+					return true;
+				}
+			}
```

Comment:
This seems to break the redirection when a template id is already set in the meta and a user tries to redirect to a unsupported product type.
It appears that when we switch to an un-supported product type we don't remove the `_product_template_id` meta field that was previously set. Maybe we should remove it? Or run the logic below first?

Reply:
Done -> https://github.com/woocommerce/woocommerce/pull/43341/commits/c76b6ac3afc5f07a25823db97212c47d0e5eabb1